Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs

```diff
@@ -22,39 +22,51 @@ public async Task<object> RunAsync(string inputJson)
 
             var akvClient = new KeyVaultClient(input.KeyId);
 
-            // Extract signature algorithm from the certificate
-            var leafCert = await akvClient.GetCertificateAsync();
-            var keySpec = leafCert.KeySpec();
-            var signatureAlgorithm = keySpec.ToSignatureAlgorithm();
-
-            // Sign
-            var signature = await akvClient.SignAsync(signatureAlgorithm, input.Payload);
-
-            // Build the certificate chain
-            List<byte[]> certificateChain = new List<byte[]>();
+            // Obtain the certificate chain
+            X509Certificate2Collection certBundle;
+            X509Certificate2 leafCert;
             if (input.PluginConfig?.ContainsKey("ca_certs") == true)
             {
-                // Build the entire certificate chain from the certificate 
-                // bundle (including the intermediate and root certificates).
-                var caCertsPath = input.PluginConfig["ca_certs"];
-                certificateChain = CertificateChain.Build(leafCert, CertificateBundle.Create(caCertsPath));
+                // Obtain the certificate bundle from file 
+                // (including the intermediate and root certificates).
+                certBundle = CertificateBundle.Create(input.PluginConfig["ca_certs"]);
+
+                // obtain the leaf certificate from Azure Key Vault
+                leafCert = await akvClient.GetCertificateAsync();
             }
-            else if (input.PluginConfig?.ContainsKey("as_secret") == true)
+            else if (input.PluginConfig?.GetValueOrDefault("as_secret")?.Equals("true", StringComparison.OrdinalIgnoreCase) == true)
```

Comment:
I'm not sure if `TryGetValue()` can also apply here or not.

Reply:
Updated.